sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1105,10 +1105,16 @@ class SessionCatalog(
    * Check if the function with the specified name exists
    */
   def functionExists(name: FunctionIdentifier): Boolean = {
-    val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
```

Comment:
The `db` is not only pass to `externalCatalog.functionExists` but also used to check.
So I think this change need more thinking.

Reply:
Best to check with @gatorsmile as this check was added on purpose in https://github.com/apache/spark/pull/12385/files Maybe it is a valid optimization.
This would need a JIRA, but first confirm this change is OK